src/items.cpp

```diff
@@ -372,502 +374,801 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
+					break;
+				}
+
+				case ITEM_PARSE_FLUIDSOURCE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					if (tmpStrValue == "water") {
+						it.fluidSource = FLUID_WATER;
+					} else if (tmpStrValue == "blood") {
+						it.fluidSource = FLUID_BLOOD;
+					} else if (tmpStrValue == "beer") {
+						it.fluidSource = FLUID_BEER;
+					} else if (tmpStrValue == "slime") {
+						it.fluidSource = FLUID_SLIME;
+					} else if (tmpStrValue == "lemonade") {
+						it.fluidSource = FLUID_LEMONADE;
+					} else if (tmpStrValue == "milk") {
+						it.fluidSource = FLUID_MILK;
+					} else if (tmpStrValue == "mana") {
+						it.fluidSource = FLUID_MANA;
+					} else if (tmpStrValue == "life") {
+						it.fluidSource = FLUID_LIFE;
+					} else if (tmpStrValue == "oil") {
+						it.fluidSource = FLUID_OIL;
+					} else if (tmpStrValue == "urine") {
+						it.fluidSource = FLUID_URINE;
+					} else if (tmpStrValue == "coconut") {
+						it.fluidSource = FLUID_COCONUTMILK;
+					} else if (tmpStrValue == "wine") {
+						it.fluidSource = FLUID_WINE;
+					} else if (tmpStrValue == "mud") {
+						it.fluidSource = FLUID_MUD;
+					} else if (tmpStrValue == "fruitjuice") {
+						it.fluidSource = FLUID_FRUITJUICE;
+					} else if (tmpStrValue == "lava") {
+						it.fluidSource = FLUID_LAVA;
+					} else if (tmpStrValue == "rum") {
+						it.fluidSource = FLUID_RUM;
+					} else if (tmpStrValue == "swamp") {
+						it.fluidSource = FLUID_SWAMP;
+					} else if (tmpStrValue == "tea") {
+						it.fluidSource = FLUID_TEA;
+					} else if (tmpStrValue == "mead") {
+						it.fluidSource = FLUID_MEAD;
+					} else {
+						std::cout << "[Warning - Items::parseItemNode] Unknown fluidSource: " << valueAttribute.as_string() << std::endl;
+					}
+					break;
+				}
+
+				case ITEM_PARSE_READABLE: {
+					it.canReadText = valueAttribute.as_bool();
+					break;
+				}
+
+				case ITEM_PARSE_WRITEABLE: {
+					it.canWriteText = valueAttribute.as_bool();
+					it.canReadText = it.canWriteText;
+					break;
+				}
+
+				case ITEM_PARSE_MAXTEXTLEN: {
+					it.maxTextLen = pugi::cast<uint16_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_WRITEONCEITEMID: {
+					it.writeOnceItemId = pugi::cast<uint16_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_WEAPONTYPE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					if (tmpStrValue == "sword") {
+						it.weaponType = WEAPON_SWORD;
+					} else if (tmpStrValue == "club") {
+						it.weaponType = WEAPON_CLUB;
+					} else if (tmpStrValue == "axe") {
+						it.weaponType = WEAPON_AXE;
+					} else if (tmpStrValue == "shield") {
+						it.weaponType = WEAPON_SHIELD;
+					} else if (tmpStrValue == "distance") {
+						it.weaponType = WEAPON_DISTANCE;
+					} else if (tmpStrValue == "wand") {
+						it.weaponType = WEAPON_WAND;
+					} else if (tmpStrValue == "ammunition") {
+						it.weaponType = WEAPON_AMMO;
```

Comment:
You should use a map instead of this if/elseif chain. It also applies to a lot of other places.